### ISSUE TITLE
mtp_draft.h: fast path (no weights) writes only 1 logit instead of full vocab_size — uninitialized memory read

### ISSUE THREAD

**Issue by bong-water-water-bong** [1bit-MONSTER/1bit-MONSTER#980]
## Summary

In `spec-decode/draft/mtp_draft.h:265-268`, the fast path when weights aren't loaded writes only `draft_logits[0]`:

```cpp
if (w_.empty()) {
    // Fast path: no trained weights yet — passthrough so callers/tests don't crash.
    for (int i = 0; i < H; i++) draft_hidden[i] = trunk_hidden[i];
    draft_logits[0] = draft_hidden[0];  // ← only 1/151936 logits written!
    return;
}
```

The comment says "so callers/tests don't crash" — but it doesn't fill the rest of the `vocab_size` (151936) logits buffer. The remaining 151,935 floats are **uninitialized**.

## Impact

- If any caller reads logits beyond index 0 with no weights loaded, they get garbage values (potential NaN, denormals, or sensitive data left on the stack/heap)
- The `SpeculativeDecoder::generate()` does check weights before calling draft, so the main path is safe — but any test or alternate code path that calls `forward()` without loaded weights gets UB
- The issue is the fast path's name is misleading: it prevents a crash from a null pointer but introduces a subtler data-dependent bug

## Suggested fix

Fill the entire logits buffer with zeros on the fast path:

```cpp
if (w_.empty()) {
    std::copy(trunk_hidden, trunk_hidden + H, draft_hidden);
    std::fill(draft_logits, draft_logits + cfg_.vocab_size, 0.0f);
    return;
}
```
